pkg/helm/helm_cli.go

```diff
@@ -11,43 +11,36 @@ import (
 	"github.com/pkg/errors"
 )
 
-type runCommandFunc func(dir string, name string, args ...string) error
-type runCommandWithOutputFunc func(dir string, name string, args ...string) (string, error)
-
-type helmRunner struct {
-	run           runCommandFunc
-	runWithOutput runCommandWithOutputFunc
-}
-
 // HelmCLI implements common helm actions based on helm CLI
 type HelmCLI struct {
 	Binary     string
 	BinVersion Version
 	CWD        string
-	runner     helmRunner
+	Runner     *util.Command
 }
 
 // NewHelmCLI creates a new HelmCLI instance configured to used the provided helm CLI in
 // the given current working directory
-func NewHelmCLI(binary string, version Version, cwd string) *HelmCLI {
-	return &HelmCLI{
-		Binary:     binary,
-		BinVersion: version,
-		CWD:        cwd,
-		runner: helmRunner{
-			run:           util.RunCommandVerbose,
-			runWithOutput: util.RunCommandWithOutput,
-		},
+func NewHelmCLI(binary string, version Version, cwd string, args ...string) *HelmCLI {
+	a := []string{}
+	for _, x := range args {
+		y := strings.Split(x, " ")
+		for _, z := range y {
+			a = append(a, z)
+		}
 	}
-}
-
-func newHelmCLIWithRunner(binary string, version Version, cwd string, runner helmRunner) *HelmCLI {
-	return &HelmCLI{
+	runner := &util.Command{
+		Args: a,
+		Name: binary,
+		Dir:  cwd,
+	}
+	cli := &HelmCLI{
 		Binary:     binary,
 		BinVersion: version,
 		CWD:        cwd,
-		runner:     runner,
+		Runner:     runner,
```

Comment:
I would not expose this outside of the package. The idea is that it can be used only by the tests to inject a fake executor, in order to monitor the command input.

Reply:
Got it. You need it for reflect. Right?

Reply:
Initially I was exposing it as I wanted to swap out in testing. Then I just patched it. I don't mind too much either way tbh.